ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/UtxoLedgerService.java

```diff
@@ -183,4 +184,53 @@ void sendUpdatedTransactionBuilder(
             @NotNull UtxoTransactionBuilder transactionBuilder,
             @NotNull FlowSession session
     );
+
+
+    /**
+     * Creates a query object for a named ledger query with the given name - if it exists. This query can be executed later.
+     * <p>
+     * Example usage:
+     * <ul>
+     * <li>Kotlin:<pre>{@code
+     * val query = utxoLedgerService.query("FIND_BY_TEST_FIELD", Int::class.java)
+     *     .setParameter("testField", "value")
+     *     .setParameter("participants", listOf("something"))
+     *     .setParameter("contractStateType", ContractState::class.java.name)
+     *     .setParameter("in-memory-filter-parameter", "parameter")
+     *     .setCreatedTimestampLimit(Instant.now())
+     *     .setOffset(0)
+     *     .setLimit(100)
+     * do {
+     *     val resultSet = query.execute()
+     *     processResultsWithApplicationLogic(resultSet.results)
+     *     query.setOffset(query.offset + 100)
```

Comment:
If you have a look at the discussion above you can see that the new offset was confusing to @driessamyn so we (@lankydan) decided to remove that

Reply:
Yeah, me and @driessamyn agreed to remove the `newOffset` concept and instead to introduce a `ResultSet.getNext()` in the future when we complete the optimisation on the db worker side to re-execute queries if the page size is not filled up. This would then understand how to manage the correct offset internally.